handler/src/test/java/io/netty/handler/ipfilter/IpSubnetFilterTest.java

```diff
@@ -141,6 +144,52 @@ public void testUniqueIpFilterHandler() {
         Assert.assertTrue(ch4.isActive());
     }
 
+    @Test
+    public void testBinarySearch() {
+        List<IpSubnetFilterRule> ipSubnetFilterRuleList = new ArrayList<IpSubnetFilterRule>();
+        ipSubnetFilterRuleList.add(buildRejectIP("1.2.3.4", 32));
+        ipSubnetFilterRuleList.add(buildRejectIP("1.1.1.1", 8));
+        ipSubnetFilterRuleList.add(buildRejectIP("200.200.200.200", 32));
+        ipSubnetFilterRuleList.add(buildRejectIP("108.0.0.0", 4));
+        ipSubnetFilterRuleList.add(buildRejectIP("10.10.10.10", 8));
```

Comment:
Would be good to also test IPv6, including testing with rule lists that is a mix of IPv4 and IPv6, and have them match on both IPv4 and IPv6 addresses.

Reply:
And here we go again, Comparator tries to compare IPv4 and IPv6. How about explicitly defining the IP version type?

Reply:
Yeah, that could work. If we ensure that the list never contain both types, it would side-step a lot of trouble.

Reply:
The comparator, or the accept method, would still need to test that the address being checked is of the correct type, though. But that should be simple enough.

Reply:
Added IPv4/6 count check. It'll throw an exception like this:
```java
java.lang.IllegalArgumentException: Mixed IP version rules are not allowed. Number of IPv4 rules: 5, IPv6 rules: 1
```

Reply:
@chrisvest PTAL at the latest code.